### PR TITLE
Add AveragePool-19 - add `dilations` attribute

### PR DESCRIPTION
**Pull request by p-wysocki**
Signed-off-by: p-wysocki [przemyslaw.wysocki@intel.com](mailto:przemyslaw.wysocki@intel.com)

### Description
Add AveragePool-19, which now has `dilations` attribute.

### Motivation and Context
- Partially solves https://github.com/onnx/onnx/issues/2290
- The issue touches on the subject of alignment of Pooling operators. Their attributes are as follows:
image
- Related PRs: https://github.com/onnx/onnx/pull/4533 and https://github.com/onnx/onnx/pull/4534

### CODE REVIEW

File: onnx/reference/ops/op_average_pool.py [hunk 1]
```diff
@@ -27,3 +28,89 @@ def _run(  # type: ignore
...
+            auto_pad=auto_pad,
+            ceil_mode=ceil_mode,
+            dilations=dilations,
+            kernel_shape=kernel_shape,
+            pads=pads,
+            strides=strides,
+        )
+
+
+if onnx_opset_version() > 11:
```
Comment (xadupre) [inline onnx/reference/ops/op_average_pool.py:109]: It should be `>= 19`. AveragePool - 11 is used from opset 11 to 18.

**Comment by xadupre** [comment]
You may remove line https://github.com/onnx/onnx/blob/main/onnx/reference/ops/_op_common_pool.py#L216 (216 to 220). They are no longer necessary. That's what causing the issue.

File: onnx/reference/ops/op_average_pool.py (hunk 1 above)
Comment (gramalingam) [reply onnx/reference/ops/op_average_pool.py:109]: Similar points for the checks below, I think?

Comment (xadupre) [reply onnx/reference/ops/op_average_pool.py:109]: Yes. I removed these lines in #4800 to make it simpler.

File: onnx/backend/test/case/node/averagepool.py [hunk 2]
```diff
@@ -455,3 +455,34 @@ def export_averagepool_2d_ceil() -> None:
...
+        node = onnx.helper.make_node(
+            "AveragePool",
+            inputs=["x"],
+            outputs=["y"],
+            kernel_shape=[3, 3],
+            strides=[1, 1],
+            dilations=[2, 2],
+            ceil_mode=True,
+        )
+        x = np.array(
```
Comment (xadupre) [inline onnx/backend/test/case/node/averagepool.py:474]: I don't know if it is better to let it that way or to replace it by `(np.arange(16) + 1).astype(np.int64).reshape((1, 1, 4, 4))`.

**Review by xadupre — APPROVED** [review]

File: onnx/backend/test/case/node/averagepool.py (hunk 2 above)
Comment (p-wysocki) [reply onnx/backend/test/case/node/averagepool.py]: Good point, done.

**Review by xadupre — APPROVED** [review]

**Review by gramalingam — APPROVED** [review]

**Pull request merged**